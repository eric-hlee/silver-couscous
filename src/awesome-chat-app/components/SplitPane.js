import React from 'react'

class SplitPane extends React.Component {

  render() {

    const style_splitPanel = {
      display: 'flex',
      flexDirection: 'row',
      height: '100%',
      border: '1px solid silver',
      overflow: 'hidden',
      borderTop: '0',
    };

    const style_splitPaneLeft = {
      border: '1px solid #555555',
      flex: 'auto',
      padding: '10px',
      width: '30%',
      minHeight: '500px',
      minWidth: '150px',
      borderRight: '0',
    }

    const style_splitPaneRight = {
      border: '1px solid #555555',
      flex: 'auto',
      padding: '10px',
      width: '70%',
      minHeight: 'auto',
      minWidth: '1024px',
      position: 'relative',
      overflow:'auto',
    }

    return (
      <div className="SplitPane" style={style_splitPanel}>
        <div className="SplitPane-left" style={style_splitPaneLeft}>
          {this.props.left}
        </div>
        <div className="SplitPane-right" style={style_splitPaneRight}>
          {this.props.right}
        </div>
      </div>
    );
  }
}

export default SplitPane