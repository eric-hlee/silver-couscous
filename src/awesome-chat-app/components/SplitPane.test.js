import React from 'react'
import { shallow } from 'enzyme'
import SplitPane from './SplitPane'

describe('SplitPane', () => {
  it('should render self and subcomponents', () => {
    const wrapper = shallow(<SplitPane />)
    expect(wrapper).toMatchSnapshot()
  })
})
