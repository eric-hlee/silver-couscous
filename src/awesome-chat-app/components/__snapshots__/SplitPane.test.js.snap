// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitPane should render self and subcomponents 1`] = `
<div
  className="SplitPane"
  style={
    Object {
      "border": "1px solid silver",
      "borderTop": "0",
      "display": "flex",
      "flexDirection": "row",
      "height": "100%",
      "overflow": "hidden",
    }
  }
>
  <div
    className="SplitPane-left"
    style={
      Object {
        "border": "1px solid #555555",
        "borderRight": "0",
        "flex": "auto",
        "minHeight": "500px",
        "minWidth": "150px",
        "padding": "10px",
        "width": "30%",
      }
    }
  />
  <div
    className="SplitPane-right"
    style={
      Object {
        "border": "1px solid #555555",
        "flex": "auto",
        "minHeight": "auto",
        "minWidth": "1024px",
        "overflow": "auto",
        "padding": "10px",
        "position": "relative",
        "width": "70%",
      }
    }
  />
</div>
`;
